sherpa/astro/io/pyfits_backend.py

```diff
@@ -78,23 +79,27 @@ def _has_hdu(hdulist, name):
     return True
 
 
-def _try_key(hdu, name, fix_type=False, dtype=SherpaFloat):
+def _try_key(hdu, name, *, fix_type=False, dtype=SherpaFloat):
 
     try:
         key = hdu.header[name]
     except KeyError:
         return None
 
+    # TODO: should this comparison be case insensitive, so that
+    # NONE and None are also matched?
+    #
```

Comment:
So, we have historically had an issue where we have had `None` values get converted to `"None`", causing endless fun later on. Or at least that's my memory of the situation and it's been long enough ago I don't have tickets/PRs to point to.

Saying that, the OGIP standard is not precise when it comes to BACKFILE/RESPFILE/ANCRFILE/CORRFILE - you have an example using "none" in https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/spectra/ogip_92_007/node12.html but I don't believe it's ever exactly specified what case this must be.

Reply:
I will add that this test is I think overly broad since it'll replace the keyword value "This is definitely not none." with None but it does match the crates backend behaviour. I don't want to make too significant a change here at this stage to "improve" this code without a more-throrough review!

Reply:
[and some of my concerns about the None/'None' handling islikely related to just the crates backend]. I'm removing the original comment and replacint it with a "TODO" comment,

Reply:
I'll leave this for @hamogu to decide whether it's been resolved or needs more discussion.